kloppy/infra/serializers/event/wyscout/deserializer_v2.py

```diff
@@ -537,19 +564,29 @@ def deserialize(self, inputs: WyscoutInputs) -> EventDataset:
                         **generic_event_args,
                     )
 
-                    if event.event_type.name == "DUEL":
-                        # when DuelEvent is interception, we need to overwrite this and the previous DuelEvent
-                        events = events[:-1]
-                        event = interception_event
-                    elif event.event_name in ["recovery", "miscontrol"]:
-                        event = interception_event
-                    elif event.event_name in ["pass", "clearance"]:
-                        events.append(
-                            transformer.transform_event(interception_event)
-                        )
-
-                if event and self.should_include_event(event):
-                    events.append(transformer.transform_event(event))
+                    for i, new_event in enumerate(list(new_events)):
+                        if new_event.event_type == EventType.DUEL:
+                            # when DuelEvent is interception, we need to
+                            # overwrite this and the previous DuelEvent
+                            events = events[:-1]
+                            new_events[i] = interception_event
+                        elif new_event.event_type in [
+                            EventType.RECOVERY,
+                            EventType.MISCONTROL,
+                        ]:
+                            # replace touch events
+                            new_events[i] = interception_event
+                        elif new_event.event_type in [
+                            EventType.PASS,
+                            EventType.CLEARANCE,
+                        ]:
+                            # insert an interception event before interception passes
+                            new_events.insert(i, interception_event)
+
+                for new_event in new_events:
+                    if self.should_include_event(new_event):
+                        events.append(transformer.transform_event(new_event))
+                new_events = []
```

Comment:
I'm probably overlooking something obvious, but it's not clear to me what the purpose of this line of code is.

Reply:
The `new_events = []` line has no purpose. I overlooked it. 😃